### PR TITLE
Misc. VLSI makefile improvements

### PR DESCRIPTION
**Pull request by abejgonzalez**
Variable-ized the makefile to removed duplication in places. Use `*.f` file to include proper blackboxes into input files. Move `common.mk` import below `default` rule so that `default` rule is actually the first rule (so `make` works correctly).

Please let me know if you like or don't like the names. I have no pref. on the names.

### CODE REVIEW

File: vlsi/Makefile [hunk 1]
```diff
@@ -14,109 +14,123 @@ sim_dir=$(abspath .)
...
-	$(ROCKET_SRC_DIR)/plusarg_reader.v \
-	$(ROCKET_SRC_DIR)/EICG_wrapper.v \
-
-ALL_RTL = $(ROCKET_SRCS) $(TOP_FILE) $(TOP_SMEMS_FILE)
+OBJ_DIR            ?= $(vlsi_dir)/build
+ENV_YML            ?= $(vlsi_dir)/bwrc-env.yml
+INPUT_CONFS        ?= example.yml $(dir $(tech_dir))/bwrc.yml
+HAMMER_EXEC        ?= ./example-vlsi
 
-CLOCK_DOMAINS = $(build_dir)/$(long_name).domains
```
Comment (abejgonzalez) [inline vlsi/Makefile]: What this used anywhere important?

**Comment by colinschmidt** [comment]
We're gonna refractor this file entirely once we switch to the hammer generated makefile fragment. So you can just merge this willy nilly until then

File: vlsi/Makefile [hunk 2]
```diff
@@ -14,109 +14,123 @@ sim_dir=$(abspath .)
...
+#########################################################################################
+# general rules
+#########################################################################################
+ALL_RTL = $(TOP_FILE) $(TOP_SMEMS_FILE) $(extra_v_includes)
```
[the remark this answers is not in the document]
Comment (abejgonzalez) [reply vlsi/Makefile]: its defined under this line (line 36)

File: vlsi/Makefile [hunk 3]
```diff
@@ -14,109 +14,123 @@ sim_dir=$(abspath .)
...
+# synthesis
+#########################################################################################
+SYN_CONF = $(OBJ_DIR)/inputs.yml
+GENERATED_CONFS = $(SYN_CONF) $(SRAM_CONF)
+ALL_CONFS = $(INPUT_CONFS) $(GENERATED_CONFS)
+SYN_OUT_VLOG = $(OBJ_DIR)/syn-rundir/$(TOP).mapped.v
+SYN_OUT_JSON = $(OBJ_DIR)/syn-rundir/syn-output.json
+PAR_IN_JSON = $(OBJ_DIR)/par-input.json
+
+$(SYN_CONF): $(ALL_RTL) $(extra_v_includes) $(sim_top_blackboxes)
```
[the remark this answers is not in the document]
Comment (abejgonzalez) [reply vlsi/Makefile]: Good catch.

File: vlsi/Makefile [hunk 4]
```diff
@@ -14,109 +14,123 @@ sim_dir=$(abspath .)
...
 	mkdir -p $(dir $@)
 	echo "synthesis.inputs:" > $@
 	echo "  top_module: $(TOP)" >> $@
 	echo "  input_files:" >> $@
-	for x in $(ALL_RTL); do \
+	for x in $(ALL_RTL) $(extra_v_includes) `cat $(sim_top_blackboxes)`; do \
```
[the remark this answers is not in the document]
Comment (abejgonzalez) [reply vlsi/Makefile]: Noted.

File: vlsi/Makefile [hunk 5]
```diff
@@ -14,109 +14,123 @@ sim_dir=$(abspath .)
...
+PAR_OUT_JSON = $(OBJ_DIR)/par-rundir/par-output.json
+DRC_IN_JSON = $(OBJ_DIR)/drc-input.json
 
 .PHONY: par place-and-route
-par: $(OBJ_DIR)/par-rundir/$(TOP).gds
+par: $(GDS)
 place-and-route: par
 
-$(OBJ_DIR)/par-rundir/$(TOP).gds $(OBJ_DIR)/par-rundir/par-output.json: $(OBJ_DIR)/par-input.json $(OBJ_DIR)/syn-rundir/$(TOP).mapped.v
+$(GDS) $(PAR_OUT_JSON): $(PAR_IN_JSON) $(SYN_OUT_VLOG)
```
[the remark this answers is not in the document]
Comment (abejgonzalez) [reply vlsi/Makefile]: What should the filename string be? `par-rundir/par-output.v`?

**Comment by abejgonzalez** [comment]
@colinschmidt whats the eta on the hammer makefrag support?

File: vlsi/Makefile (hunk 5 above)
[the remark this answers is not in the document]
Comment (jwright6323) [reply vlsi/Makefile]: `$(OBJ_DIR)/par-rundir/$(TOP).lvs.v`

**Comment by colinschmidt** [comment]
I think its ready to go assuming everyone on dev is ok with it being in development. We used it on the last chip(so its silicon proven ;) ) and its indispensable once you go hierarchical.

**Comment by jwright6323** [comment]
@abejgonzalez @colinschmidt should we just take over this PR to do the hammer-build integration?

**Comment by abejgonzalez** [comment]
I think that makes sense. I think it being in `dev` is fine for now.

**Comment by abejgonzalez** [comment]
Going to close this since @colinschmidt @jwright6323 took over this work.

**Pull request closed without merging**